hooks/useIsLargeScreen.ts

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react';
+
+// Checks if the screen width is greater than 1024px
+export const useIsLargeScreen = () => {
+  const [isLargeScreen, setIsLargeScreen] = useState(false);
+
+  useEffect(() => {
+    const checkScreenSize = () => {
+      setIsLargeScreen(window.innerWidth > 1024);
+    };
+
+    checkScreenSize(); // Check on initial render
+
+    window.addEventListener('resize', checkScreenSize);
+    return () => window.removeEventListener('resize', checkScreenSize);
+  }, []);
+
+  return isLargeScreen;
+};
```

Comment:
```suggestion
import { useMemo } from 'react';

// Checks if the screen width is greater than 1024px
export const useIsLargeScreen = () => {
  const isLargeScreen = useMemo(() => window.innerWidth > 1024, [window.innerWidth])

  return isLargeScreen;
};
```
I believe this is the same performance-wise and doesn't add a screen listener.